.github/workflows/tests.yml

```diff
@@ -1,10 +1,14 @@
 name: 'Tests'
 
 on:
-  push:
```

Comment:
we'll have to enforce "update with main" before merge if we remove it. Can you put it in another PR so we can merge the rest while this point is discussed.